### PR TITLE
jetpack: Remove last uses of gulp

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

Previous PRs have reduced the use of gulp to being just a wrapper around
webpack. Removing use of gulp in favor of calling webpack directly lets
us remove 161 JS dependencies from the monorepo, several of them
long-obsolete versions since gulp's devs are morally opposed to updating
deps if they don't have to.

Also, clean up `.gitattributes` in a few other projects that still had
reference to long-deleted gulpfiles.

#### Jetpack product discussion

None

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* There should be no material changes to the built artifacts. Just changelogs, the `.gitattributes` changes, version bumps, and the comment in `class.jetpack.php`.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -10758,25 +10612,6 @@ packages:
       parse5-htmlparser2-tree-adapter: 6.0.1
       tslib: 2.3.1
 
-  /chokidar/2.1.8:
-    resolution: {integrity: sha512-ZmZUazfOzf0Nve7duiCKD23PFSCs4JPoYyccjUFF3aQkQadqBhfzhjkwBH2mNOG9cTBwhamM37EIsIkZw3nRgg==}
-    deprecated: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
```
Comment (anomiex) [inline pnpm-lock.yaml:10763]: :tada:

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -14126,33 +13762,13 @@ packages:
...
     dev: true
 
   /fs.realpath/1.0.0:
     resolution: {integrity: sha1-FQStJSMVjKpA20onh8sBQRmU6k8=}
 
-  /fsevents/1.2.13:
-    resolution: {integrity: sha512-oWb1Z6mkHIskLzEJ/XWX0srkpkTQ7vaopMQkyaEIoq0fmtFVxOthb8cCxeT+p3ynTdkk/RZwbgG4brR5BeWECw==}
-    engines: {node: '>= 4.0'}
-    os: [darwin]
-    deprecated: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
```
Comment (anomiex) [inline pnpm-lock.yaml:14148]: :tada:

**Review by sdixon194 — APPROVED** [review]
CI is happy, Brad is happy, ✅

**Comment by anomiex** [comment]
r243411-wpcom

**Pull request merged**